pkg/binding/binding.go

```diff
@@ -32,24 +32,26 @@ import (
 )
 
 // syncBinding syncs a binding object with what is resolved by the bindingpolicy resolver.
-func (c *Controller) syncBinding(ctx context.Context, key util.Key) error {
+func (c *Controller) syncBinding(ctx context.Context, objIdentifier util.ObjectIdentifier) error {
+	logger := klog.FromContext(ctx)
+
 	var unstructuredObj *unstructured.Unstructured
-	if !c.bindingPolicyResolver.ResolutionExists(key.NamespacedName.Name) {
+	if !c.bindingPolicyResolver.ResolutionExists(objIdentifier.ObjectName.Name) {
 		// if a resolution is not associated to the binding's name
 		// then the bindingpolicy has been deleted, and the binding
 		// will eventually be garbage collected. We can safely ignore this.
 
 		return nil
 	}
 
-	obj, err := c.getObjectFromKey(key)
+	obj, err := c.getObjectFromObjectIdentifier(objIdentifier)
 	if errors.IsNotFound(err) {
-		unstructuredObj = util.EmptyUnstructuredObjectFromKey(key)
+		// a resolution exists and the object is not found, therefore it is deleted and should be created
+		unstructuredObj = util.EmptyUnstructuredObjectFromIdentifier(objIdentifier)
```

Comment:
FYI: Since this func is defined in the same package as the type of its first argument, this func could be a method on that type of object instead.